### ISSUE TITLE
Integrator already bound to context issue

### ISSUE THREAD

**Issue by pgrinaway** [choderalab/openmmtools#289]
Using the `MCMCMove.apply()` paradigm, I'm seeing this error. You can find my subclass `MCMCMove` [here](https://github.com/choderalab/perses/pull/396/files#diff-8c7c21f6a47e02cedcdc16233c702142R31), so if anything looks anomalous, please do let me know. 

```
[2017-09-11 19:22:29,771: ERROR/PoolWorker-3] Task perses.distributed.feptasks.run_protocol[1fad75d6-0797-4e8e-a317-9a516dec3b1f] raised unexpected: Exception('This Integrator is already bound to a context',)
Traceback (most recent call last):
  File "/Users/grinawap/anaconda/lib/python3.5/site-packages/openmmtools-0.13.1-py3.5.egg/openmmtools/cache.py", line 405, in get_context
    context = self._lru[context_id]
  File "/Users/grinawap/anaconda/lib/python3.5/site-packages/openmmtools-0.13.1-py3.5.egg/openmmtools/cache.py", line 146, in __getitem__
    entry = self._data.pop(key)
KeyError: (-8522523064050290135, -6901986784493923843)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/grinawap/anaconda/lib/python3.5/site-packages/celery/app/trace.py", line 367, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/Users/grinawap/anaconda/lib/python3.5/site-packages/celery/app/trace.py", line 622, in __protected_call__
    return self.run(*args, **kwargs)
  File "/Users/grinawap/PycharmProjects/perses/perses/distributed/feptasks.py", line 209, in run_protocol
    ne_mc_move.apply(thermodynamic_state, sampler_state)
  File "/Users/grinawap/anaconda/lib/python3.5/site-packages/openmmtools-0.13.1-py3.5.egg/openmmtools/mcmc.py", line 657, in apply
    context, integrator = context_cache.get_context(thermodynamic_state, integrator)
  File "/Users/grinawap/anaconda/lib/python3.5/site-packages/openmmtools-0.13.1-py3.5.egg/openmmtools/cache.py", line 407, in get_context
    context = thermodynamic_state.create_context(integrator, self._platform)
  File "/Users/grinawap/anaconda/lib/python3.5/site-packages/openmmtools-0.13.1-py3.5.egg/openmmtools/states.py", line 817, in create_context
    return openmm.Context(system, integrator)
  File "/Users/grinawap/anaconda/lib/python3.5/site-packages/simtk/openmm/openmm.py", line 6262, in __init__
    this = _openmm.new_Context(*args)
Exception: This Integrator is already bound to a context

**Comment by pgrinaway** [comment]
Probably best to discuss in person, since the reproduction of this bug/error is rather involved.

**Comment by jchodera** [comment]
Isn't this just missing a deepcopy of `integrator` (which goes through a serialized XML to create an integrator unbound to any context) before creating the new context in [this code block](https://github.com/choderalab/openmmtools/blob/master/openmmtools/cache.py#L399-L408)?

**Comment by andrrizzi** [comment]
This was fixed in #291.
